### PR TITLE
adding latest changes

### PR DESCRIPTION
**Pull request by absap**
update ADT_Backend to the lastest implementation.
every further PRs will be smaller and more function specific

### CODE REVIEW

File: src/zcl_abapgit_res_repos.clas.xml [hunk 1]
```diff
@@ -3,14 +3,15 @@
  <asx:abap xmlns:asx="http://www.sap.com/abapxml" version="1.0">
   <asx:values>
    <VSEOCLASS>
-    <CLSNAME>ZCL_ABAPGIT_RES_REPOS</CLSNAME>
+    <CLSNAME>CL_ABAPGIT_RES_REPOS</CLSNAME>
```
Comment (larshp) [inline src/zcl_abapgit_res_repos.clas.xml]: wrong name

Comment (larshp) [inline src/zcl_abapgit_res_repos.clas.xml]: wrong name

File: src/zcx_adt_rest_abapgit.clas.abap [hunk 2]
```diff
@@ -13,7 +13,7 @@ class ZCX_ADT_REST_ABAPGIT definition
         !IX_ERROR       type ref to CX_ROOT
         !IV_HTTP_STATUS type I optional
       raising
-        ZCX_ADT_REST_ABAPGIT .
+        CX_ADT_REST_ABAPGIT .
```
Comment (larshp) [inline src/zcx_adt_rest_abapgit.clas.abap]: wrong name

File: src/zcx_adt_rest_abapgit.clas.abap [hunk 3]
```diff
@@ -138,7 +138,7 @@ CLASS ZCX_ADT_REST_ABAPGIT IMPLEMENTATION.
       IX_ERROR     = LX_ERROR
       IV_ATTRIBUTE = LS_MSG_KEY-ATTR4 ).
 
-    raise exception type ZCX_ADT_REST_ABAPGIT
+    raise exception type CX_ADT_REST_ABAPGIT
```
Comment (larshp) [inline src/zcx_adt_rest_abapgit.clas.abap]: wrong name

File: src/zabapgit_repos.enho.xml [hunk 4]
```diff
@@ -9,7 +9,7 @@
     <ENH_BADI_IMPL_DATA>
      <SPOT_NAME>SADT_REST_RFC_APPLICATION</SPOT_NAME>
      <BADI_NAME>BADI_ADT_REST_RFC_APPLICATION</BADI_NAME>
-     <IMPL_NAME>ZABAPGIT_REPOS_RES_APPL</IMPL_NAME>
+     <IMPL_NAME>ABAPGIT_REPOS_RES_APPL</IMPL_NAME>
```
Comment (larshp) [inline src/zabapgit_repos.enho.xml]: wrong name?

File: src/zabapgit_repos.enho.xml [hunk 5]
```diff
@@ -50,7 +50,7 @@
     <ENH_BADI_IMPL_DATA>
      <SPOT_NAME>SADT_REST_RFC_APPLICATION</SPOT_NAME>
      <BADI_NAME>BADI_ADT_DISCOVERY_PROVIDER</BADI_NAME>
-     <IMPL_NAME>ZABAPGIT_REPOS_DISCOVERY</IMPL_NAME>
+     <IMPL_NAME>ABAPGIT_REPOS_DISCOVERY</IMPL_NAME>
```
Comment (larshp) [inline src/zabapgit_repos.enho.xml]: name

File: src/zcl_abapgit_res_repo_info_ext.clas.abap [hunk 6]
```diff
@@ -1,102 +1,108 @@
...
         ENDIF.
 
-*** Prepare Response
+        "Prepare Response
         response->set_body_data( content_handler = lo_response_content_handler data = ls_response_data ).
         response->set_status( cl_rest_status_code=>gc_success_ok ).
 
       CATCH zcx_abapgit_exception INTO DATA(lx_abapgit_exception).
-        zcx_adt_rest_abapgit=>raise_with_error( lx_abapgit_exception ).
+        cx_adt_rest_abapgit=>raise_with_error( lx_abapgit_exception ).
```
Comment (larshp) [inline src/zcl_abapgit_res_repo_info_ext.clas.abap]: wrong name

File: src/zcl_abapgit_res_repo_info_ext.clas.testclasses.abap [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+    add_xml '  <user>b</user>'.
+    add_xml '  <password>c</password>'.
+    add_xml '</repository_ext>'.
+    CALL TRANSFORMATION id SOURCE XML lv_xml RESULT XML lv_input_xml.
+
+    APPEND INITIAL LINE TO lt_result ASSIGNING <lt_result>.
+    <lt_result>-name = zcl_abapgit_res_repo_info_ext=>co_root_name_request.
+    GET REFERENCE OF ls_data INTO <lt_result>-value.
+
+    CALL TRANSFORMATION abapgit_st_repo_info_ext_req
```
Comment (larshp) [inline src/zcl_abapgit_res_repo_info_ext.clas.testclasses.abap]: wrong name?

File: src/zcl_abapgit_res_repo_info_ext.clas.testclasses.abap [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+    add_xml '  <password>c</password>'. "exchanged by USER
+    add_xml '  <user>b</user>'. "exchanged by PASSWORD
+    add_xml '</repository_ext>'.
+    CALL TRANSFORMATION id SOURCE XML lv_xml RESULT XML lv_input_xml.
+
+    APPEND INITIAL LINE TO lt_result ASSIGNING <lt_result>.
+    <lt_result>-name = zcl_abapgit_res_repo_info_ext=>co_root_name_request.
+    GET REFERENCE OF ls_data INTO <lt_result>-value.
+
+    CALL TRANSFORMATION abapgit_st_repo_info_ext_req
```
Comment (larshp) [inline src/zcl_abapgit_res_repo_info_ext.clas.testclasses.abap]: wrong name?

File: src/zcl_abapgit_res_repo_info_ext.clas.testclasses.abap [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+    add_xml '  <user>b</user>'.
+    add_xml '  <password>c</password>'.
+    add_xml '</repository_ext>'.
+    CALL TRANSFORMATION id SOURCE XML lv_xml RESULT XML lv_input_xml.
+
+    APPEND INITIAL LINE TO lt_result ASSIGNING <lt_result>.
+    <lt_result>-name = zcl_abapgit_res_repo_info_ext=>co_root_name_request.
+    GET REFERENCE OF ls_data INTO <lt_result>-value.
+
+    CALL TRANSFORMATION abapgit_st_repo_info_ext_req
```
Comment (larshp) [inline src/zcl_abapgit_res_repo_info_ext.clas.testclasses.abap]: wrong name

**Comment by larshp** [comment]
I have a few comments, but I guess we are more or less forced to merge this once it compiles. So I will open separate issues

File: src/zcl_abapgit_res_repos.clas.locals_imp.abap [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+  METHOD lif_abapgit_provider~validate_transport_request.
+
+    DATA: lv_error_message TYPE string.
+
+    SELECT SINGLE * FROM e070 INTO @DATA(ls_e070)
+      WHERE
+      trkorr = @iv_transport_request.
+
+    IF sy-subrc NE 0.
+      MESSAGE e003(a4c_agit_adt) WITH iv_transport_request INTO lv_error_message.
```
Comment (larshp) [inline src/zcl_abapgit_res_repos.clas.locals_imp.abap]: wrong name

File: src/zcl_abapgit_res_repos.clas.locals_def.abap [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+    set_authentication_info IMPORTING iv_user     TYPE string
+                                      iv_password TYPE string,
+    perform_import IMPORTING is_request_data TYPE zcl_abapgit_res_repos=>ty_request_data
+                   RAISING   zcx_abapgit_exception.
+ENDINTERFACE.
+CLASS lzcl_abapgit_provider DEFINITION.
+  PUBLIC SECTION.
+    INTERFACES: lif_abapgit_provider.
+ENDCLASS.
+CLASS ltzcl_abapgit_repos_resource DEFINITION DEFERRED.
```
Comment (larshp) [inline src/zcl_abapgit_res_repos.clas.locals_def.abap]: this looks like testing code, so it should be moved to the testing include

**Comment by larshp** [comment]
I've approved the review, but not checked the details, guess we should just merge the PR and then move from there

**Comment by absap** [comment]
I'm not that huge fan of try and error, but hopefully, within the next PRs we should be on a clean track

**Pull request merged**